src/ModelValidationAsyncActionFilter.cs

```diff
@@ -97,22 +124,26 @@ private bool ShouldIgnoreFilter(ActionExecutingContext context)
             return !hasApiControllerAttribute;
         }
 
-        private async Task ValidateEnumerableObjectsAsync(object value, ModelStateDictionary modelState)
+        private async Task ValidateEnumerableObjectsAsync(object value,
+            ModelStateDictionary modelState)
         {
             var underlyingType = value.GetType().GenericTypeArguments[0];
             var validator = GetValidator(underlyingType);
 
             if (validator == null)
                 return;
 
-            foreach (var item in (IEnumerable)value)
+            foreach (var item in (IEnumerable) value)
             {
                 if (item is null)
                     continue;
                 var context = new ValidationContext<object>(item);
                 var result = await validator.ValidateAsync(context);
 
-                result.AddToModelState(modelState, string.Empty);
+                var errorCode = result.Errors?.FirstOrDefault()?.ErrorCode;
```

Comment:
**question**: The order of the error entries affects the result? If the first validation that fails is a "normal" one and the second one has an `.WithErrorCode(ErrorCode.Forbidden);` what would be the the error code?